File: Motocitizen/src/motocitizen/utils/popups/AccidentListPopup.java
package motocitizen.utils.popups;

import android.content.Context;
import android.widget.PopupWindow;

import motocitizen.content.Content;
import motocitizen.content.accident.Accident;
import motocitizen.dictionary.Medicine;
import motocitizen.user.User;
import motocitizen.utils.DateUtils;
import motocitizen.utils.Preferences;
import motocitizen.utils.Utils;
//todo renew after hide/end/activate

public class AccidentListPopup extends PopupWindowGeneral {
    private final Accident point;
    private final String   accText;

    public AccidentListPopup(Context context, int id) {
        super(context);
        point = Content.INSTANCE.accident(id);
        accText = getAccidentTextToCopy(point);
    }

    public PopupWindow getPopupWindow(Context context) {
        content.addView(copyButtonRow(context, accText));
        for (String phone : Utils.getPhonesFromText(point.getDescription())) {
            content.addView(phoneButtonRow(context, phone), layoutParams);
            content.addView(smsButtonRow(context, phone), layoutParams);
        }
        if (User.INSTANCE.isModerator() || Preferences.INSTANCE.getLogin().equals(point.ownerName()))
            content.addView(finishButtonRow(point));

        if (User.INSTANCE.isModerator()) {
            content.addView(hideButtonRow(point));
            content.addView(banButtonRow(context, point.getOwner()), layoutParams);
        }

        content.addView(shareMessage(context, accText));
        content.addView(coordinatesButtonRow(context, point), layoutParams);
        popupWindow.setContentView(content);
        return popupWindow;
    }

    public static String getAccidentTextToCopy(Accident accident) {
        StringBuilder res = new StringBuilder();
        res.append(DateUtils.getDateTime(accident.getTime())).append(" ");
        res.append(accident.ownerName()).append(": ");
        res.append(accident.getType().getText()).append(". ");
        if (accident.getMedicine() != Medicine.UNKNOWN) {
            res.append(accident.getMedicine().getText()).append(". ");
        }
        res.append(accident.getAddress()).append(". ");
        res.append(accident.getDescription()).append(".");
        return res.toString();
    }
}

File: Motocitizen/src/motocitizen/utils/popups/AccidentListPopup.kt
package motocitizen.utils.popups

import android.content.Context
import android.widget.PopupWindow
import motocitizen.content.accident.Accident
import motocitizen.dictionary.Medicine
import motocitizen.user.User
import motocitizen.utils.Preferences
import motocitizen.utils.dateTimeString
import motocitizen.utils.getAccidentTextToCopy
import motocitizen.utils.getPhonesFromText

//todo renew after hide/end/activate

class AccidentListPopup(val context: Context, private val accident: Accident) : PopupWindowGeneral(context) {
    fun getPopupWindow(context: Context): PopupWindow {
        val accText = accident.getAccidentTextToCopy()

        content.addView(copyButtonRow(context, accText))
        for (phone in getPhonesFromText(accident.description)) {
            content.addView(phoneButtonRow(context, phone), layoutParams)
            content.addView(smsButtonRow(context, phone), layoutParams)
        }
        if (User.isModerator || Preferences.login == accident.ownerName())
            content.addView(finishButtonRow(accident))

        if (User.isModerator) {
            content.addView(hideButtonRow(accident))
            content.addView(banButtonRow(context, accident.owner), layoutParams)
        }

        content.addView(shareMessage(context, accText))
        content.addView(coordinatesButtonRow(context, accident), layoutParams)
        popupWindow.contentView = content
        return popupWindow
    }
}
